remote-test: use u64 to represent file size

onur-ozkan opened the pull request:
Currently, triggering a transfer of data exceeding the size of 4294967295 bytes results in a panic on the `remote-test-server` as `io::copy(&mut file, dst) failed with Connection reset by peer (os error 104)`. This issue happens because the size is transmitted as u32 to `remote-test-server`.

First commit increases the supported file size. But I am not sure about its necessity — can we realistically encounter file sizes exceeding 4GB in builds, perhaps through some complicated configurations?
 
~The second commit adds a sanity check to avoid encountering the error `io::copy(&mut file, dst) failed with Connection reset by peer (os error 104)` on the `remote-test-server` side.~

onur-ozkan commented:
> First commit increases the supported file size. But I am not sure about its necessity — can we realistically encounter file sizes exceeding 4GB in builds, perhaps through some complicated configurations?

Let me know if this isn't needed (so I can revert).

Mark-Simulacrum commented:
I don't think we need to support large files, though I also don't mind supporting them.

That said, if we're touching this code I think rather than manually bit twiddling using https://doc.rust-lang.org/nightly/std/primitive.u32.html#method.to_le_bytes and correspondign from_le_bytes feels much better to me.

onur-ozkan committed:
use u64 to represent file size

This increases the maximum supported file size (previously limited to 4GB)
and avoids potential issues with u32 to u64 conversions, which are no longer
needed.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

onur-ozkan commented:
> The second commit adds a sanity check to avoid encountering the error io::copy(&mut file, dst) failed with Connection reset by peer (os error 104) on the remote-test-server side.

Just realized that there is no need for a sanity check on the size anymore since we transmit the size without any conversion.

Updated the manual bit-twiddling part to use `to_be_bytes` on the client side (server side already uses `from_be_bytes`).

@rustbot ready

Mark-Simulacrum commented on src/tools/remote-test-client/src/main.rs:
```diff
@@ -300,18 +300,18 @@ fn run(support_lib_count: usize, exe: String, all_args: Vec<String>) {
+        let mut amt = 0u64;
+        for (i, &byte) in header[1..=8].iter().enumerate() {
+            amt |= (byte as u64) << (56 - 8 * i);
+        }
```
`u64::from_be_bytes(header[1..9].try_into().unwrap())`

Mark-Simulacrum commented:
@bors r+

bors commented:
:pushpin: Commit 1afd216f760601fb13a9982d283ef8e2224b7527 has been approved by `Mark-Simulacrum`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=Mark-Simulacrum 1afd216f760601fb13a9982d283ef8e2224b7527 -->
<!-- homu: {"type":"Approved","sha":"1afd216f760601fb13a9982d283ef8e2224b7527","approver":"Mark-Simulacrum","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
